leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/servlet/json/SecurityDeserializer.java

```
@@ -102,6 +109,77 @@ public SecurityInfo deserialize(JsonElement json, Type typeOfT, JsonDeserializat
                 info = SecurityInfo.newRawPublicKeyInfo(endpoint, key);
             } else if (x509 != null && x509.getAsBoolean()) {
                 info = SecurityInfo.newX509CertInfo(endpoint);
+            } else if (oscore != null) {
+                // OSCORE Deserialization
+
+                // Parse hexadecimal context parameters
+                byte[] masterSecret = Hex.decodeHex(oscore.get("masterSecret").getAsString().toCharArray());
+                byte[] senderId = Hex.decodeHex(oscore.get("senderId").getAsString().toCharArray());
+                byte[] recipientId = Hex.decodeHex(oscore.get("recipientId").getAsString().toCharArray());
```

should we raise JSON exception is this mandatory field are absent ?